src/IrcConnection/IrcConnection.cs

```diff
@@ -1254,11 +923,12 @@ private void _Worker()
 #if LOG4NET
                         Logger.Socket.Warn("connection lost");
 #endif
+                        // CHECK: shouldn't be necessary any longer
                         // only flag this as connection error if we are not
                         // cleanly disconnecting
-                        if (!_Connection.IsDisconnecting) {
-                            _Connection.IsConnectionError = true;
-                        }
+                        //if (!_Connection.IsDisconnecting) {
+                        //    _Connection.IsConnectionError = true;
+                        //}
```

Comment:
IsConnectionError doesn't exist in IrcConnection anymore. It's handled by the transports. If we are cleanly disconnecting, the original code doesn't set IsConnectionError and neither does the new transport. If we are not cleanly disconnecting, the transport will have already set Transport.IsConnectionError = true by the time we get here, therefore the code should no longer be necessary. I cannot offer you a 100% cast-iron guarantee but that is how it generally works.

How do you want to implement testing for this?